pkg/registry/apis/provisioning/resources/tree.go

```diff
@@ -25,13 +25,16 @@ type FolderTree interface {
 	AddUnstructured(item *unstructured.Unstructured) error
 	Count() int
 	Walk(ctx context.Context, fn WalkFunc) error
+	HasFailed(folder string) bool
```

Comment:
The tree should agnostic to its usage. That behavior is something, we should implement on top.